src/mono/wasm/debugger/BrowserDebugHost/Startup.cs

```diff
@@ -112,6 +85,23 @@ string GetEndpoint(HttpContext context)
                     return $"{devToolsHost.Scheme}://{devToolsHost.Authority}{request.Path}{request.QueryString}";
                 }
 
+                async Task ReturnPageLinksToDebug(HttpContext context)
+                {
+                    HttpRequest request = context.Request;
+                    Dictionary<string, string>[] tabs = await ProxyGetJsonAsync<Dictionary<string, string>[]>(GetEndpoint(context) + "json/list");
+                    context.Response.ContentType = "text/html";
+                    string urlsToInspect = "<title>Inspectable pages</title><h3>Inspectable pages</h3><hr>";
+                    if (tabs.Length == 0)
+                        urlsToInspect += "No inspectable pages available";
+                    foreach (var tab in tabs)
+                    {
+                        string aHref = tab["devtoolsFrontendUrl"];
+                        aHref = aHref.Replace($"ws={devToolsHost.Authority}", $"ws={request.Host}");
+                        urlsToInspect += $"<a href=\"http://{devToolsHost.Authority}{aHref}\">{tab["title"]}<br><span>{tab["url"]}</a><br><br>";
+                    }
+                    await context.Response.Body.WriteAsync(Encoding.ASCII.GetBytes(urlsToInspect));
+                }
+
                 async Task Copy(HttpContext context)
                 {
                     using (var httpClient = new HttpClient { Timeout = TimeSpan.FromSeconds(5) })
```

Comment:
not for this PR, but we should use a static instance of `HttpClient`. From https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-6.0#remarks :

```
HttpClient is intended to be instantiated once and re-used throughout the life of an application. Instantiating an HttpClient class for every request will exhaust the number of sockets available under heavy loads. This will result in SocketException errors. 
```